mono/utils/mono-threads-coop.c

```diff
@@ -67,10 +77,36 @@ mono_threads_state_poll (void)
 	}
 }
 
+static void *
+return_stack_ptr ()
+{
+	int i;
+	return &i;
+}
+
+static void
+copy_stack_data (MonoThreadInfo *info, void* stackdata_begin, void* stackdata_end)
```

Comment:
This is the function that should have __builtin_unwind_init.

It should not have a stackdata_end argument as it won't include the saved registers. Just add a local variable and use its address.